crates/typst-library/src/text/quote.rs

```diff
@@ -0,0 +1,209 @@
+        Self::Label(label) => label.into_value(),
+    },
+    content: Content => Self::Content(content),
+    label: Label => Self::Label(label),
+}
+
+impl Show for QuoteElem {
+    fn show(&self, vt: &mut Vt, styles: StyleChain) -> SourceResult<Content> {
+        let mut realized = self.body();
+        let block = self.block(styles);
+
+        if self.quotes(styles) == Smart::Custom(true) || !block {
+            // use h(0pt, weak: true) to make the quotes "sticky"
+            let quote = SmartquoteElem::new().with_double(true).pack();
+            let weak_h = HElem::new(Spacing::Rel(Rel::zero())).with_weak(true).pack();
+
+            realized = Content::sequence([
+                quote.clone(),
+                weak_h.clone(),
+                realized,
+                weak_h,
+                quote,
+            ]);
+        }
+
+        if block {
+            realized = BlockElem::new().with_body(Some(realized)).pack();
+
+            if let Some(attribution) = self.attribution(styles) {
+                let mut seq = vec![TextElem::packed('—'), SpaceElem::new().pack()];
+
+                match attribution {
+                    Attribution::Content(content) => {
+                        seq.push(content);
+                    }
+                    Attribution::Label(label) => {
+                        let citation = vt.delayed(|vt| {
+                            let citation = CiteElem::new(vec![label.0]);
+                            let bib =
+                                BibliographyElem::find(vt.introspector).at(self.span())?;
+
+                            // TODO: these should use the citation-format attribute, once CSL
+                            // is implemented and retreive the authors for non-author formats
+                            // themeselves, see:
+                            // - https://github.com/typst/typst/pull/2252#issuecomment-1741146989
+                            // - https://github.com/typst/typst/pull/2252#issuecomment-1744634132
+                            Ok(match bib.style(styles) {
+                                // author-date and author
+                                BibliographyStyle::Apa
+                                | BibliographyStyle::Mla
+                                | BibliographyStyle::ChicagoAuthorDate => {
+                                    citation.with_brackets(false).pack()
+                                }
+                                // notes, label and numeric
+                                BibliographyStyle::ChicagoNotes
+                                | BibliographyStyle::Ieee => citation.pack(),
+                            })
+                        });
+
+                        seq.push(citation);
+                    }
+                }
+
+                // use v(0.9em, weak: true) bring the attribution closer to the quote
+                let weak_v = VElem::weak(Spacing::Rel(Em::new(0.9).into())).pack();
```

Comment:
Is this value customizable without redefining the whole quote?

Reply:
No, it is not, but I hardly see someone adjusting only this and not the whole quote. `0.9` was chosen by @reknih. Adding a parameter for just that doesn't sound right, it's very much tied to this specific default show implementation.

Reply:
No, a parameter for this does indeed not sound right. Just wondering. I guess it's fine. I have just noticed how as soon as there's any hardcoded number or separator in a default show rule, somebody opens an issue about customization.